src/components/search/index.tsx

```diff
@@ -51,10 +54,14 @@ export default function Search({
             <input className={style.input} {...getInputProps()} />
           </div>
 
-          <GeolocationButton
-            className={style.geolocationButton}
-            onGeolocate={selectItem}
-          />
+          {onSwap ? (
+            <SwapButton className={style.geolocationButton} onSwap={onSwap} />
+          ) : (
+            <GeolocationButton
+              className={style.geolocationButton}
+              onGeolocate={selectItem}
+            />
+          )}
```

Comment:
I think maybe a better approach here is to be more generic in the API. Pass inn a button component from parent for instance. 🤔

Reply:
Good idea! I've changed this now 👍